iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/SlackTerminalImpl.java

```diff
@@ -31,6 +32,37 @@ public class SlackTerminalImpl extends AbstractMultiVariantIdentifiableExtension
         this.setTerminal(terminal);
     }
 
+    private void unregisterReferencedTerminalIfNeeded(int variantIndex) {
+        // check there is no more same terminal referenced by any variant, unregister it
+        Terminal oldTerminal = terminals.get(variantIndex);
+        if (oldTerminal != null && !terminals.contains(oldTerminal)) {
```

Comment:
similarly to ReferenceTerminals, always false

Reply:
Fixed